qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +163,163 @@ def _add_variable(self,
                       upperbound: Union[float, int] = INFINITY,
                       vartype: VarType = VarType.CONTINUOUS,
                       name: Optional[str] = None) -> Variable:
-        """Checks whether a variable name is already taken and adds the variable to list and index
-        if not.
+        return cast(List[Variable], self._add_variables([], _update_var_list, lowerbound,
+                                                        upperbound, vartype, name))[0]
+
+    def _add_variables(self,
+                       initial_structure: Union[List, Dict[str, Variable]],
+                       update_function,
+                       lowerbound: Union[float, int] = 0,
+                       upperbound: Union[float, int] = INFINITY,
+                       vartype: VarType = VarType.CONTINUOUS,
+                       name: Optional[str] = None,
+                       formatter: Optional[str] = None,
+                       keys: Union[int, Sequence] = 1) -> Union[List[Variable],
+                                                                Dict[str, Variable]]:
+        if isinstance(keys, int) and keys < 1:
+            raise QiskitOptimizationError(
+                "Cannot create non-positive number of variables: {}".format(keys))
+        if not name and formatter:
+            name = 'x'
+        if name and not formatter:
+            if isinstance(keys, int) and keys > 1 or isinstance(keys, Sequence) and len(keys) > 0:
+                formatter = '{}'
+            else:
+                formatter = ''
+                if name in self._variables_index:
+                    raise QiskitOptimizationError("Variable name already exists: {}".format(name))
+        if not name and not formatter:
+            name = 'x'
+            formatter = '{}'
+        if '{{}}' in formatter:
+            raise QiskitOptimizationError(
+                "Formatter cannot contain nested substitutions: {}".format(formatter))
+        substitution_count = formatter.count('{}')
+        if substitution_count > 1:
+            raise QiskitOptimizationError(
+                "Formatter cannot contain more than one substitution: {}".format(formatter))
+        return self._create_var_struct_update_maps(lowerbound, upperbound, vartype, name,
+                                                   formatter, keys, initial_structure,
+                                                   update_function)
+
+    def _create_var_struct_update_maps(self,
```

Comment:
It looks a bit too tricky. Please take a look at my impl.
https://github.com/JosDenmark/qiskit-aqua/pull/1

Reply:
Your changes have been incorporated.